regression/pages/common/utils.py

```diff
@@ -0,0 +1,526 @@
+    return required_cookies
+
+
+def get_coupon_request(target_url, cookie_str):
+    """
+    Send a Get request and return the response if successful
+    Args:
+      target_url:
+      cookie_str:
+    """
+    kwargs = {}
+    headers = {
+        'Accept': 'text/html,application/xhtml+xml,'
+                  'application/xml;q=0.9,image/webp,*/*;q=0.8',
+        'Accept-Language': 'en-US,en;q=0.8',
+        'Accept-Encoding': 'gzip, deflate, sdch, br',
+        'Cookie': cookie_str
+    }
+    kwargs.update({
+        'headers': headers,
+        'auth': HTTPBasicAuth(AUTH_USERNAME, AUTH_PASSWORD),
+    })
+    response = requests.get(target_url, **kwargs)
+    if response.ok:
+        return response.content
+    else:
+        msg = "Could not use restful API.  Status code: {0}".format(
+            response.status_code)
+        raise ApiException(msg)
+
+
+def get_coupons_from_email(coupon_string):
+    """
+    Read coupon codes and urls from email
+    Ignore first two lines of file as these are headers
+    Args:
+        coupon_string:
+    Returns:
+        coupon codes and urls:
+    """
+    coupons = {}
+    for i, row in enumerate(coupon_string.splitlines()):
+        if i > 3:
+            if row:
+                new_row = row.split(',')
+                coupons.update({new_row[0]: new_row[1]})
+    return coupons
+
+
+def skip_cleanup_if_test_passed():
+    """
+    Skip cleanup when tests are passing
+    """
+    def decorator(test_function):
+        """
+        Decorator
+        Args:
+            test_function:
+        """
+        @functools.wraps(test_function)
+        def wrapper(self, *args, **kwargs):
+            """
+            Wrapper
+            Args:
+                self:
+                *args:
+                **kwargs:
+            """
+            test_function(self, *args, **kwargs)
+            self.full_cleanup = False
+        return wrapper
+    return decorator
+
+
+def fill_input_fields(page, css_selectors, values):
+    """
+    A helper function to fill multiple fields in a form
+    Args:
+        page:
+        css_selectors:
+        values:
+    """
+    for css_selector, value in izip(css_selectors, values):
+        page.q(css=css_selector).fill(value)
+
+
+def select_value_from_drop_down(page, drop_down_name, value):
+    """
+    A helper function to select value from drop down
+    Args:
+        page:
+        drop_down_name:
+        value:
+    """
+    page.wait_for_element_visibility(
+        'select[name=' + drop_down_name + ']', 'Drop down is visible')
+    page.q(
+        css='select[name=' + drop_down_name + '] option[value="{}"]'.format(
+            value)).click()
+    page.wait_for(
+        lambda: page.q(
+            css='select[name=' + drop_down_name + '] option[value="{}"]'.
+            format(value)).selected,
+        "Correct value is selected",
+        timeout=DEFAULT_TIMEOUT
+    )
+
+
+def select_values_from_drop_downs(page, drop_down_names, values):
```

Comment:
You basically rewrote the above function with a loop. Instead, you can loop on calls to `select_value_from_drop_down`.